src/app/message.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MessageService {
```

Comment:
This is starting to look like a store :) 

I guess a good candidate to move to ngrx when you get to that part.